### PR TITLE
Fixes #1042:RPM repo still uses HTTP, not HTTPS

### LINKED ISSUE #1042: RPM repo still uses HTTP, not HTTPS
For example, the sample YUM file at https://docs.projectcalico.org/v2.5/getting-started/openstack/installation/redhat uses HTTP for the repo.

## Expected Behavior
Should be using HTTPS everywhere.

## Current Behavior
HTTP used for YUM.

## Possible Solution
May be enough to simply do `s/http/https/` in those URLs.

### PR DESCRIPTION
**Pull request by rakrup**
## Description
Type of fix : Bug Fix
Reason : Fixes bug in documentation 
Link to the issue this PR fixes : https://github.com/projectcalico/calico/issues/1042

### CODE REVIEW

**Comment by rakrup** [comment]
the Test succeeds in the local environment but fails at semaphoreci. Below are logs from my machine.

```
Running ["LinkCheck", "ImageCheck", "HtmlCheck", "ScriptCheck"] on ["/_site"] on *.html... 
Checking 1214 external links...
Ran on 952 files!
HTML-Proofer finished successfully.
docker run -ti -e JEKYLL_UID=`id -u` --rm -v $PWD/_site:/_site/ quay.io/calico/htmlproofer /_site --assume-extension --check-html --empty-alt-ignore --url-ignore "#"
Running ["LinkCheck", "ImageCheck", "HtmlCheck", "ScriptCheck"] on ["/_site"] on *.html... 


Checking 1674 external links...
make: [htmlproofer] Error 137 (ignored)
# Rerun htmlproofer across _all_ files, but ignore failure, allowing us to notice legacy docs issues without failing CI
docker run -v $PWD:/calico --entrypoint /bin/sh -ti garethr/kubeval:0.1.1 -c 'find /calico/_site/master -name "*.yaml" |grep -v config.yaml | xargs /kubeval'
Unable to find image 'garethr/kubeval:0.1.1' locally
0.1.1: Pulling from garethr/kubeval
e110a4a17941: Pull complete 
90e467a1ca31: Pull complete 
514b4bc8c47d: Pull complete 
Digest: sha256:0c8716cec21c30642815cdce5ec14cedee98d876d6e268b2ecdd5dc84cf45142
Status: Downloaded newer image for garethr/kubeval:0.1.1
The document /calico/_site/master/getting-started/kubernetes/installation/hosted/calico.yaml is a valid ConfigMap
The document /calico/_site/master/getting-started/kubernetes/installation/hosted/calicoctl.yaml is a valid Pod
The document /calico/_site/master/getting-started/kubernetes/installation/hosted/kubeadm/1.5/calico.yaml is a valid ConfigMap
The document /calico/_site/master/getting-started/kubernetes/installation/hosted/kubeadm/1.6/calico.yaml is a valid ConfigMap
The document /calico/_site/master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/calico-networking/1.7/calico.yaml is a valid ConfigMap
The document /calico/_site/master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/calicoctl.yaml is a valid Pod
The document /calico/_site/master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.7/calico.yaml is a valid ConfigMap
The document /calico/_site/master/getting-started/kubernetes/installation/hosted/rbac-kdd.yaml is a valid ClusterRole
The document /calico/_site/master/getting-started/kubernetes/installation/manifests/skydns.yaml is a valid Service
The document /calico/_site/master/getting-started/kubernetes/installation/policy-controller.yaml is a valid Deployment
The document /calico/_site/master/getting-started/kubernetes/installation/rbac.yaml is a valid ClusterRole
The document /calico/_site/master/getting-started/kubernetes/tutorials/stars-policy/manifests/00-namespace.yaml is a valid Namespace
The document /calico/_site/master/getting-started/kubernetes/tutorials/stars-policy/manifests/01-management-ui.yaml is a valid Namespace
The document /calico/_site/master/getting-started/kubernetes/tutorials/stars-policy/manifests/02-backend.yaml is a valid Service
The document /calico/_site/master/getting-started/kubernetes/tutorials/stars-policy/manifests/03-frontend.yaml is a valid Service
The document /calico/_site/master/getting-started/kubernetes/tutorials/stars-policy/manifests/04-client.yaml is a valid Namespace
The document /calico/_site/master/getting-started/kubernetes/tutorials/stars-policy/policies/allow-ui-client.yaml is a valid NetworkPolicy
The document /calico/_site/master/getting-started/kubernetes/tutorials/stars-policy/policies/allow-ui.yaml is a valid NetworkPolicy
The document /calico/_site/master/getting-started/kubernetes/tutorials/stars-policy/policies/backend-policy.yaml is a valid NetworkPolicy
The document /calico/_site/master/getting-started/kubernetes/tutorials/stars-policy/policies/default-deny.yaml is a valid NetworkPolicy
The document /calico/_site/master/getting-started/kubernetes/tutorials/stars-policy/policies/frontend-policy.yaml is a valid NetworkPolicy
Rahuls-MacBook-Pro:calico rakrup$ 
```

**Comment by emanic** [comment]
@rakrup Thank you very much for your contribution! We will review this carefully. Apologies for the issues with the Contributor's license agreement and the discrepancies between the local htmlproofer and the Semaphore job. We will update CONTRIBUTING_DOCS.md to clarify these areas. Thanks again and we will get back to you soon.

**Comment by fasaxc** [comment]
+1 for this change but it looks like the binaries server isn't responding over HTTPS right now, I'll get someone to look into that.

**Comment by neiljerram** [comment]
binaries.projectcalico.org is responding on HTTPS now, so I think that means that this PR could be merged - @fasaxc ?

File: _includes/version_warning.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {% if page.version != site.data.versions.first[0] %}
   <div class="alert alert-warning" role="alert">
-    <strong>Warning:</strong> You are viewing the {{ page.version }} docs. <a href="{{site.baseurl}}/latest">Click here to view the docs for the latest release.</a>
+	  <strong>Warning:</strong> You are viewing the {{ page.version }} docs. <a href="{{site.baseurl}}/latest/{{page.url | remove:page.version }}">Click here to view the docs for the latest release.</a>
```
Comment (fasaxc) [inline _includes/version_warning.html]: Please can you pull this change out into a separate PR and update the 404 page so that, if the document doesn't exist in /latest/ the user doesn't get confused?

File: v2.5/getting-started/openstack/installation/redhat.md [hunk 2]
```diff
@@ -71,11 +71,11 @@ Configure the Calico repository:
     cat > /etc/yum.repos.d/calico.repo <<EOF
     [calico]
     name=Calico Repository
-    baseurl=http://binaries.projectcalico.org/rpm/{{ ppa_repo_name }}/
+    baseurl=https://binaries.projectcalico.org/rpm/{{ ppa_repo_name }}/
```
Comment (fasaxc) [inline v2.5/getting-started/openstack/installation/redhat.md]: Just spotted that you made this change in the `/v2.5/` directory which is the fork of our docs for the v2.5 release.  Please can you copy it to `/master/`?  (We maintain all the versions in one branch of the repo so we can easily back-apply changes.)

File: _includes/version_warning.html (hunk 1 above)
Comment (rakrup) [reply _includes/version_warning.html]: My bad ! this was a fix to another bug. Let me re-organise it to another branch and open a separate PR for the same. Realised after pushing.

Comment (rakrup) [reply _includes/version_warning.html]: The fix to the other bug was pushed only today so you couldn't have spotted it earlier @fasaxc

File: v2.5/getting-started/openstack/installation/redhat.md (hunk 2 above)
Comment (rakrup) [reply v2.5/getting-started/openstack/installation/redhat.md]: Let me look into this and fix this one

**Comment by rakrup** [comment]
Was trying to squash all the commits for the checkin together and resulted into a series of commits being added to this pull request. Is it normal ? any help regarding this ?

**Comment by fasaxc** [comment]
What command did you run, exactly?

Given how small the change was, the easiest way to fix is to cut a new branch from master and then reapply your changes manually.

**Comment by rakrup** [comment]
I was trying a rebase. Will probably what you suggested

**Comment by rakrup** [comment]
Opening a single clean commit based pull request for the issue. Closing this one.
